src/Compression/CompressionCodecZSTDQAT.cpp

```diff
@@ -0,0 +1,114 @@
+#ifdef ENABLE_ZSTDQAT_COMPRESSION
+#include <CompressionCodecZSTD.h>
+#include <Compression/CompressionInfo.h>
+#include <Compression/CompressionFactory.h>
+#include <zstd.h>
+#include <Parsers/IAST.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/ASTFunction.h>
+#include <Common/typeid_cast.h>
+#include <Poco/Logger.h>
+#include <Common/logger_useful.h>
+#include <qatseqprod.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int CANNOT_COMPRESS;
+    extern const int ILLEGAL_SYNTAX_FOR_CODEC_TYPE;
+    extern const int ILLEGAL_CODEC_PARAMETER;
+}
+
+class CompressionCodecZSTDQAT : public CompressionCodecZSTD
```

Comment:
Most of settings, which affect compression is being defined on query/session level than table level.
Plus there is ongoing PR, which implements ability to set those setting value per column. 
https://github.com/ClickHouse/ClickHouse/pull/55201

Reply:
QuickAssist-accelerates ZSTD has some limitations as below:
- QAT only supports compression levels L1 to L12
- QAT don't support ZSTD_c_enableLongDistanceMatching cParam
So i don't think it's proper to define a SETTINGS use_intel_qat = 1 under software ZSTD, otherwise we need exclude too much exceptional cases. For now ,it would be better to define ZSTDQAT new codec and enable it with the 'enable_zstd_qat_codec' setting